src/components/graph/collapse.helper.js

```diff
@@ -111,6 +111,24 @@ function getTargetLeafConnections(rootNodeId, linksMatrix = {}, { directed }) {
     }, []);
 }
 
+/**
+ * Filter out leaf connections of nodes which are not collapsible
+ * @param  {Object.<string, Object>} nodes - an object containing all nodes mapped by their id.
+ * @param {Array.<Object.<string, string>>} connections a list of leaf connections.
+ * @returns {Array.<Object.<string, string>>} a list of leaf nodes which are collapsible
+ * @memberof Graph/collapse-helper
+ */
+function filterNonCollapsibleNodes(nodes, connections = []) {
+    return connections.filter(connection => {
+        console.log(connection.target);
+        console.log(nodes[connection.target]);
+        const leafNode = nodes[connection.target];
+
+        return Object.hasOwnProperty.call(leafNode, "collapsible") ? leafNode.collapsible : true;
```

Comment:
In case `leafNode.collapsible` is `null` or `undefined` this will eval to `false` making the node noncollapsible. To make this retro-compatible, we need to assume that the absence of value on `leafNode.collapsible` means that they are collapsible. This needs to be well documented as well, it's not obvious.

I think we would need to refactor this, into something like:

```javascript
const isNodeBlockingCollapsible = leafNode?.collapsible === false;

return !isNodeBlockingCollapse;
```